sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java

```diff
@@ -84,6 +85,61 @@ protected void cleanUpResources() {
         }
     }
 
+    @Test
+    public void canCreateVMSSWithPlan() {
+        final String vmssName = generateRandomResourceName("vmss", 10);
+        ResourceGroup resourceGroup = this.resourceManager.resourceGroups().define(rgName).withRegion(region).create();
+        final String uname = "jvuser";
+
+        Network network =
+            this
+                .networkManager
+                .networks()
+                .define(generateRandomResourceName("vmssvnet", 15))
+                .withRegion(region)
+                .withExistingResourceGroup(resourceGroup)
+                .withAddressSpace("10.0.0.0/28")
+                .withSubnet("subnet1", "10.0.0.0/28")
+                .create();
+
+        PurchasePlan plan = new PurchasePlan()
+            .withName("access_server_byol")
+            .withPublisher("openvpn")
+            .withProduct("openvpnas");
+
+        ImageReference imageReference = new ImageReference()
+            .withPublisher("openvpn")
+            .withOffer("openvpnas")
+            .withSku("access_server_byol")
+            .withVersion("latest");
+
+        VirtualMachineScaleSet virtualMachineScaleSet =
+            this
+                .computeManager
+                .virtualMachineScaleSets()
+                .define(vmssName)
+                .withRegion(region)
+                .withExistingResourceGroup(resourceGroup)
+                .withSku(VirtualMachineScaleSetSkuTypes.STANDARD_A0)
+                .withExistingPrimaryNetworkSubnet(network, "subnet1")
```

Comment:
Usually it is easier to call `withNewNetwork`?

Reply:
I refer to other tests to add the network configuration part, will try withNewNetWork(). Thanks